storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -162,8 +186,19 @@ lsn_t Redo_Log_Reader::read_log_seg(log_t &log, byte *buf, lsn_t start_lsn,
 
   do {
     os_offset_t source_offset;
-
-    source_offset = file->offset(start_lsn);
+    if (log.m_files.ctx().m_files_ruleset == Log_files_ruleset::CURRENT) {
+      source_offset = file->offset(start_lsn);
```

Comment:
@altmannmarcelo for pre 8030 files, cant we populate a file object with m_start_lsn ? Then same logic should work ?

Reply:
ie ib_logfile0 & ib_logfile1 map to Log_file objects with m_start_lsn. Is this possible or not?

Reply:
@satya-bodapati as per our slack conversation and research, the problem comes when we remove redo logs and let the server re-create them. start_lsn is calculated different in this case on pre 8.0.30 and we cannot use `file->offset(lsn)` function to get the correct file offset.